src/core/annotation.js

```diff
@@ -497,16 +503,39 @@ class AnnotationBorderStyle {
    *
    * @public
    * @memberof AnnotationBorderStyle
-   * @param {integer} width - The width
+   * @param {integer} width - The width.
+   * @param {Array} rect - The annotation `Rect` entry.
    */
-  setWidth(width) {
+  setWidth(width, rect = [0, 0, 0, 0]) {
+    if (typeof PDFJSDev === 'undefined' ||
+        PDFJSDev.test('!PRODUCTION || TESTING')) {
+      assert(Array.isArray(rect) && rect.length === 4,
+             'A valid `rect` parameter must be provided.');
+    }
+
     // Some corrupt PDF generators may provide the width as a `Name`,
     // rather than as a number (fixes issue 10385).
     if (isName(width)) {
       this.width = 0; // This is consistent with the behaviour in Adobe Reader.
       return;
     }
     if (Number.isInteger(width)) {
+      if (width > 0) {
+        const maxWidth = (rect[2] - rect[0]) / 2;
+        const maxHeight = (rect[3] - rect[1]) / 2;
+
+        // Truncate ridiculously large `width`s to the `Rect` entry of the
+        // annotation, to prevent the `annotationLayer` from rendering it
+        // over the surrounding document (fixes bug1552113.pdf).
+        //
+        // This is consistent with the behaviour in Adobe Reader.
+        if ((maxWidth > 0 && maxHeight > 0) &&
+            (width > maxWidth || width > maxHeight)) {
+          warn('AnnotationBorderStyle.setWidth - truncating large width: ' +
+               width);
+          width = Math.min(maxWidth, maxHeight);
```

Comment:
A somewhat related problem, which this patch probably ought to fix as well, is that these kind of Annotations aren't currently clickable.
It appears that only the "contents", and not the border itself, is part of the clickable area and in cases such as these that may becomes zero because of https://github.com/mozilla/pdf.js/blob/209e42043af86f54bae02c37d320350bf4f96a0c/src/display/annotation_layer.js#L175-L176

The simplest solution may actually be to outright just ignore the border `width`, i.e. setting it to zero, for the cases where it's clearly wrong; suggestions on all of this is much appreciated here!

Reply:
Having thought about this some more, I'm starting to favour the "set `width = 1` (i.e. the default value)"-approach in these cases.
Besides obviously fixing the issues with non-clickable links, it still displays at least *some* visible border which seems appropriate since the PDF file did ask for a border to be displayed (albeit a ridiculously large one).

Reply:
I noticed that Okular displays a similar large blue box and it looks like the Chrome web PDF viewer doesn't display the border at all, so the behavior differs. In the specification I could only find the sentence `If present, the border shall be drawn completely inside the annotation  rectangle` and I think that's also what you want to achieve in this version of the patch. However, in my opinion the border is still way too large and causes not only the clickable region to be invisible, but also the underlying superscript text as well. Therefore, I agree that if the width is too large, it's best to set it to the default width of 1 because, as you indicated as well, there should be some border. According to the spec, only if the width is 0 no border should be drawn, so I think that's another reason to not set it to 0 in this case, but to 1.

Reply:
> How does Adobe Reader render this?

The same way as this patch does, i.e. a blue rectangle limited in size to the Annotation `Rect` (thus obscuring *only* the superscript text). However, it's at least clickable in Adobe Reader.

I'll update the patch to use `width = 1` instead in this case, and this sort of problem ought to very rare in practice; thank you!